configd/src/apps/sentinel/service.cpp

```diff
@@ -99,6 +107,13 @@ Service::terminate(bool catchable)
             return ret;
         } else {
             setState(KILLING);
+            char pstackCmd[256];
+            LOG(info, "%s:%d failed to stop. Will dump the stack",  name().c_str(), _pid);
+            snprintf(pstackCmd, sizeof(pstackCmd), "pstack %d > %s/%s.pstack.%d", _pid, getVespaTempDir().c_str(), name().c_str(), _pid);
```

Comment:
pstackCmd might not be NUL terminated, i.e. when return value from snprintf() >= sizeof(pstackCmd).
In that case, it might be best to just skip the system() call